app/src/main/kotlin/org/eclipse/kuksa/companion/feature/settings/view/EditableTextSetting.kt

```diff
@@ -0,0 +1,247 @@
+            Text(
+                text = value,
+                fontSize = SettingSecondaryFontSize,
+            )
+        }
+    }
+
+    if (isDialogOpen) {
+        EditTextDialog(
+            label = label,
+            initialValue = value,
+            keyboardType = keyboardType,
+            onDismissRequest = {
+                isDialogOpen = false
+            },
+            onClickOk = onValueChanged,
+        )
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalComposeUiApi::class)
+@Composable
+fun EditTextDialog(
+    label: String,
+    initialValue: String,
+    modifier: Modifier = Modifier,
+    keyboardType: KeyboardType = KeyboardType.Text,
+    onDismissRequest: () -> Unit = {},
+    onClickCancel: () -> Unit = {},
+    onClickOk: (String) -> Unit = {},
+) {
+    val containerColor = MaterialTheme.colorScheme.secondaryContainer
+
+    val keyboardController = LocalSoftwareKeyboardController.current
+
+    val focusRequester = remember { FocusRequester() }
+    LaunchedEffect(Unit) {
+        focusRequester.requestFocus()
+    }
+
+    val direction = LocalLayoutDirection.current
+    var textFieldValue by remember {
+        val selection = if (direction == LayoutDirection.Ltr) { // move Cursor to end
+            TextRange(initialValue.length)
+        } else {
+            TextRange.Zero
+        }
+
+        mutableStateOf(TextFieldValue(initialValue, selection))
+    }
+
+    Dialog(
+        onDismissRequest = onDismissRequest,
+        properties = DialogProperties(
+            dismissOnClickOutside = true,
+            dismissOnBackPress = true,
+        ),
+    ) {
+        Box(modifier) {
+            val interactionSource = remember { MutableInteractionSource() }
+            Column(
+                Modifier
+                    .background(containerColor, RoundedCornerShape(15.dp))
+                    .padding(25.dp),
+            ) {
+                Text(
+                    text = label,
+                    fontSize = SettingPrimaryFontSize,
+                    fontWeight = FontWeight.Bold,
+                )
+
+                Spacer(modifier = Modifier.height(10.dp))
+
+                BasicTextField(
+                    value = textFieldValue,
+                    onValueChange = { newValue ->
+                        textFieldValue = newValue
+                    },
+                    textStyle = TextStyle(
+                        fontSize = SettingSecondaryFontSize,
+                    ),
+                    keyboardOptions = KeyboardOptions(
+                        imeAction = ImeAction.Done,
+                        keyboardType = keyboardType,
+                    ),
+                    keyboardActions = KeyboardActions(
+                        onDone = {
+                            keyboardController?.hide()
+                            onDismissRequest()
+                            onClickOk(textFieldValue.text)
+                        },
+                    ),
```

Comment:
First I thought about parametrizing KeyboardOptions to make it even more flexible, however that would have forced me to also parametrize keyboardActions (due to the hard dependency to ImeAction.Done/"onDone") and therefore pull up code to where it doesn't belong. For now I intentionally decided against this and only exposed the keyboardType. 

I don't see any additional value (for now) of being able to change the ImeAction to anything other.